src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -0,0 +1,133 @@
+use anyhow::Result;
+
+use crate::{HypervisorConfig, VcpuThreadIds};
+use kata_types::capabilities::{Capabilities, CapabilityBits};
+
+const VSOCK_SCHEME: &str = "vsock";
+const VSOCK_AGENT_CID: u32 = 3;
+const VSOCK_AGENT_PORT: u32 = 1024;
+
+unsafe impl Send for QemuInner {}
+unsafe impl Sync for QemuInner {}
+
+pub struct QemuInner {
+    config: HypervisorConfig,
+}
+
+impl QemuInner {
+    pub fn new() -> QemuInner {
+        QemuInner {
+            config: Default::default(),
+        }
+    }
+
+    pub(crate) async fn prepare_vm(&mut self, _id: &str, _netns: Option<String>) -> Result<()> {
+        info!(sl!(), "Preparing QEMU VM");
+        Ok(())
```

Comment:
Why not `todo!()` here? [This comment applies to `prepare_vm()`

Reply:
That's because this version of the qemu driver has nothing to do here, it's fine with nothing being done in prepare_vm().

Reply:
Curious. The DragonBall variant does some non-trivial stuff, like setting up the VM path. Why would you not need that for `qemu`?

Reply:
I assume I will need that at some point, just not yet.  Frankly a more pressing part of `prepare_vm()` implementation at the moment is vsock CID allocation which is essential for a sandbox startup and so far only works for me if I hardcode the CID. I'm sure I'll get to the rest in due time. The immediate goal at the moment being just having a very simple container to launch, my next step is handling sandbox persistence.